core-relations/src/free_join/plan.rs

```diff
@@ -157,6 +167,88 @@ pub(crate) struct Plan {
     pub atoms: Arc<DenseIdMap<AtomId, Atom>>,
     pub stages: JoinStages,
 }
+impl Plan {
+    pub(crate) fn to_report(&self, symbol_map: &SymbolMap) -> egglog_reports::Plan {
```

Comment:
I think this could be pretty expensive. Can we cache the output of this method somehow? Or make report generation optional?